Source/NSBrowser.m

```diff
@@ -3306,68 +3388,88 @@ - (void) _performLoadOfColumn: (NSInteger)column
       [matrix setDrawsBackground: YES];
 
       if (!_allowsMultipleSelection)
-        {
-          [matrix setMode: NSRadioModeMatrix];
-        }
+	{
+	  [matrix setMode: NSRadioModeMatrix];
+	}
       [matrix setTarget: self];
       [matrix setAction: @selector(doClick:)];
       [matrix setDoubleAction: @selector(doDoubleClick:)];
-      
+
       // set new col matrix and release old
       [bc setColumnMatrix: matrix];
       RELEASE (matrix);
     }
   [sc setDocumentView: matrix];
 
-  // Loading is different based upon passive/active delegate
-  if (_passiveDelegate)
+  // Loading is different based upon item/passive/active delegate
+  if (_itemBasedDelegate)
+    {
+      // Iterate over the children for the item....
+      for (i = 0; i < rows; i++)
+	{
+	  id aCell = [matrix cellAtRow: i column: 0];
+	  if (![aCell isLoaded])
+	    {
+	      BOOL leaf = YES;
+	      id val = nil;
+
+	      child = [_browserDelegate browser: self child: i ofItem: item];
```

Comment:
We never make sure that the delegate actually supports all these methods.

Reply:
From the test, it seems that when I comment out the code for various methods in this type of delegate that it doesn't throw an exception.

Reply:
This sounds like they always check which methods are implemented.

Reply:
Hmm...  I'm a bit confused.   Like I said... all it does when I comment the code out for the implementations of these in the AppDelegate it just doesn't get the data.   I am guessing if these aren't implemented it doesn't set something that causes the browser to get the data.  I'll change the code to reflect this.

Reply:
I believe the ones that are needed are
```
- (NSInteger) browser: (NSBrowser *)browser
numberOfChildrenOfItem: (id)item;
- (id) browser: (NSBrowser *)browser
         child: (NSInteger)index
        ofItem: (id)item;
- (BOOL) browser: (NSBrowser *)browser
      isLeafItem: (id)item;
```
I will make the changes to check for these.

Reply:
I believe this is the check that is happening.  On macOS the data simply doesn't load.   This change does the same thing.